### PR TITLE
Performance optimizations

### PR DESCRIPTION
**Pull request by dbkaplun**
I was able to consolidate two loops and turn them into one... twice. So 4 loops are now 2. This is approximately 10% faster. Probably easiest to review these commits one-by-one rather than looking at the whole diff. No other optimizations are obvious to me right now.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Looks good. Array methods like `.map`/`.reduce`/etc comes with some overhead as they call a function on each iteration. Might be even faster to just use `for` loops.

File: index.js [hunk 1]
```diff
@@ -29,40 +29,32 @@ module.exports = function (str, opts) {
...
 
-	var splits = [];
-
-	for (var i = 0; i < stats.length; i++) {
-		// found the separator on the same index on all lines
-		if (stats[i] === lines.length) {
-			splits.push(i);
+	var consecutive = false;
+	var splits = stats.reduce(function (splits, occurrenceCount, pos) {
+		if (occurrenceCount !== lines.length) consecutive = false;
```
Comment (kevva) [inline index.js:34]: `if (occurrenceCount !== lines.length) {`

File: index.js [hunk 2]
```diff
@@ -29,40 +29,32 @@ module.exports = function (str, opts) {
...
+		else {
```
Comment (kevva) [inline index.js:35]: `} else if (pos !== 0 && !consecutive) {`

File: index.js [hunk 3]
```diff
@@ -29,40 +29,32 @@ module.exports = function (str, opts) {
...
-	}
-
-	var ret = lines.map(function (line, lineIndex) {
-		return headers.reduce(function (ret, header, colIndex) {
-			var el = line[colIndex] || '';
-			ret[header] = opts.transform ? opts.transform(el, header, colIndex, lineIndex) : el;
-			return ret;
-		}, {});
-	});
+		if (lineNumber === 0) headers = headers || cells;
```
Comment (kevva) [inline index.js:49]: `if (lineNumber === 0) {`

File: index.js [hunk 4]
```diff
@@ -29,40 +29,32 @@ module.exports = function (str, opts) {
...
+		else {
```
Comment (kevva) [inline index.js:50]: `} else {`

File: index.js (hunk 1 above)
Comment (dbkaplun) [inline index.js:34]: @kevva you reviewed the wrong PR bro! New PR is #5

**Pull request closed without merging**